src/graphql/models/User.js

```diff
@@ -17,13 +18,29 @@ const currentUserOnlyField = (type, resolver) => ({
   },
 });
 
+/**
+ * Use Gravatar as avatar provider.
+ */
+const avatarResolver = (s = 80, d = 'mp', r = 'g') => ({
+  type: GraphQLString,
+  description: 'return hash based on user email for gravatar url',
+  resolve(user) {
+    const GRAVATAR_URL = 'https://www.gravatar.com/avatar/';
+    const hash = md5(
+      (user.email || 'anonymous@cofacts.g0v.tw').trim().toLocaleLowerCase()
+    );
+    const params = `?s=${s}&d=${d}&r=${r}`;
```

Comment:
Let's add `&d=identicon` so that a random image will be generated when the hash does not match any registered Gravatar user.
https://en.gravatar.com/site/implement/images/